### ISSUE TITLE
azurerm_lb_rule set to backend_port set to zero

### ISSUE THREAD

**Issue by jonathanlackey** [terraform-providers/terraform-provider-azurerm#1811]
### Community Note
I am having issues  with adding an HA load balancer rule to an azurerm_lb_rule resource. I am leveraging All as the Protocol with  the back_end and front_end port set to zero. I had this issue after upgrading my azurerm provider to V1.13.0. This exact same snippet worked previously. It seems like the validator may have changed.  

Downgrading to the following code allows the snippet to work as expected:
Terraform v0.11.8
+ provider.azurerm v1.10.0

### Terraform Version
Terraform v0.11.8
+ provider.azurerm v1.13.0

### Affected Resource(s)

azurerm_lb_rule

### Terraform Configuration Files

```hcl
resource "azurerm_lb_rule" "HUB-LB1HaRule" {
  name                           = "HUB-LB1HaRule"
  resource_group_name            = "${azurerm_resource_group.HUB-RG.name}"
  loadbalancer_id                = "${azurerm_lb.HUB-LB1.id}"
  protocol = "All"
  frontend_port = "0"
  backend_port = "0"
  frontend_ip_configuration_name = "${lookup(azurerm_lb.HUB-LB1.frontend_ip_configuration[0],"name")}"
  backend_address_pool_id        = "${azurerm_lb_backend_address_pool.HUB-LB1-IpPool.id}"
  probe_id                       = "${azurerm_lb_probe.HUB-LB1-Probe.id}"
}
```

### Expected Behavior

+ azurerm_lb_rule.HUB-LB1HaRule
      id:                                                                    <computed>
      backend_address_pool_id:                                               "${azurerm_lb_backend_address_pool.HUB-LB1-IpPool.id}"
      backend_port:                                                          "0"
      enable_floating_ip:                                                    "false"
      frontend_ip_configuration_id:                                          <computed>
      frontend_ip_configuration_name:                                        "${lookup(azurerm_lb.HUB-LB1.frontend_ip_configuration[0],\"name\")}"
      frontend_port:                                                         "0"
      idle_timeout_in_minutes:                                               <computed>
      load_distribution:                                                     <computed>
      loadbalancer_id:                                                       "${azurerm_lb.HUB-LB1.id}"
      name:                                                                  "HUB-LB1HaRule"
      probe_id:                                                              "${azurerm_lb_probe.HUB-LB1-Probe.id}"
      protocol:                                                              "all"
      resource_group_name:                                                   "HUB-RG"
### Actual Behavior
Error: azurerm_lb_rule.AZE1-HUB-LB1HaRule: "backend_port" is not a valid port number: '\x00'
Error: azurerm_lb_rule.AZE1-HUB-LB1HaRule: "frontend_port" is not a valid port number: '\x00'

### Steps to Reproduce

1. `terraform apply or terraform plan'

### Important Factoids

Running in Azure East

### References

* #0000

**Comment by torstenstern** [comment]
Hello, is there any workaround for that?

**Comment by jonathanlackey** [comment]
I hard coded the version that worked for me in the provider and that took care of it in mycase:

provider "azurerm"{
  version="=v1.10.0"
...
}

**Comment by JunyiYi** [comment]
Hi @jonathanlackey @torstenstern , thanks for reporting the issue. It has been fixed in `v1.16.0` (#1951), please update to the latest provider.
